smartfile/__init__.py

```diff
@@ -124,8 +128,31 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        return self.post('/path/oper/remove', path=deletefile)
+
+    def upload(self, usrfile):
+        if usrfile[0].endswith('/'):
+            raise ValueError("File name should have no trailing slash")
+        return self.post('/path/data/', file=usrfile)
+
+    def download(self, file_to_be_downloaded):
+        """ file_to_be_downloaded is a file-like object that has already
+        been uploaded, you cannot download folders """
```

Comment:
This variable name is a bit long. I would not advocate variable names this long as it makes me type more than I should have to :-). Also, the comment is a bit confusing, as the user is starting a download of a remote file, not a file-like-object. A FLO is basically an IO stream in Python.